tests/backends/test_pep517.py

```diff
@@ -31,6 +34,20 @@ def repo_root():
     return str(Path(__file__).parents[2])
 
 
+@pytest.fixture
+def relocate_ac_index(repo_root, tmpdir):
```

Comment:
Maybe we can repurpose this fixture to be `preserve_ac_index_state`? Mainly we should also probably handle the case of removing the AC index if it never existed in the repository to start.